### DISCUSSION TITLE
Bad Lighthouse performance scores on v 9.5.1

### DISCUSSION THREAD

**Discussion by BernardA** [vercel/next.js#16308] [Help]
I am getting very bad results on performance at Lighthouse.

In principle it looks that all bad metrics are related to Nextjs as I seem to be very much by the book.
Only item that seem I can directly address is `rel="preconnect"` which I will do on next deployment.

Here is how it looks: 

![Screenshot 2020-08-18 at 14 38 25](https://user-images.githubusercontent.com/9449665/90514060-ec8d1380-e160-11ea-801e-dce289b3772a.png)

## Describe the bug

The app is available at 
https://quiamo-client.vercel.app/ 

and I also left the repository public at 
https://github.com/BernardA/quiamo-client

## To Reproduce

To reproduce just access above URL with Chrome and apply Lighthouse.

## System information

- OS: macOS
- Browser Chrome
- Version of Next.js: 9.5.1

## Additional context

The app is deployed at Vercel

**Comment by prateekbh** [comment]
Hi!
Is the lighthouse score taken on your machine  or on webpage test?

**Reply by BernardA** [reply]
Just visit https://quiamo-client.vercel.app/ with Chrome, open Chrome's dev tools/Lighthouse

**Reply by prateekbh** [reply]
got it, Lets use webpage test's LH score for a reference here so that machine/network difference are out of the way

**Reply by prateekbh** [reply]
Can you try `optimizeFonts` experimental flag with 9.5.2?

**Reply by mkhoussid** [reply]
Where can we find a list of available experimental flags? Couldn't find it anywhere

**Comment by BernardA** [comment]
Update available with 9.5.2. 

As I was expecting that did not help much, if at all.  The issue is clearly bigger than just fonts.

That said I do have an issue with `<link rel="preload" crossorigin`.
I am looking into it. 
![Screenshot 2020-08-19 at 13 40 08](https://user-images.githubusercontent.com/9449665/90630643-b1064e00-e221-11ea-897f-2f23a65da92b.png)

**Comment by prateekbh** [comment]
So this score was yesterday before 9.5.2: https://www.webpagetest.org/result/200818_TH_500c9c2b3639f6cab47c69994bd7c082/1/details/#waterfall_view_step1

**Reply by BernardA** [reply]
Not sure what your point is. 
It was bad yesterday with 9.5.1 and still as bad with 9.5.2.

**Reply by prateekbh** [reply]
sorry this was just a reference as I try to find things step by step

**Comment by BernardA** [comment]
@prateekbh, Any news on this? 
This score is very important for my clients - and thus for me as well.
I am selling  Nextjs as being great for performance, but this type of result does not rhyme with that argument.

Also, not sure why you moved this from issues. Why  are you assuming this is not an actual issue?

**Comment by prateekbh** [comment]
@BernardA apologies, I have been a bit occupied last week.
Here's a trace from your current deployed version: https://www.webpagetest.org/result/200901_8X_05bdb245a32e60f4fa4370337d0e40a7/

Here's the light house breakup: 
https://googlechrome.github.io/lighthouse/scorecalc/#FCP=2240&SI=3652&LCP=4282&TTI=4591&TBT=1201&CLS=0.93&FCI=4591&FMP=2240&device=none&version=6.3.0

Few low hanging fruits here as observed [here](https://www.webpagetest.org/video/compare.php?tests=200901_8X_05bdb245a32e60f4fa4370337d0e40a7-r:1-c:0):
- There is a major CLS @ 5.5s tile having ~5points deduction from score
- Logo seems to be downloading as an image, holding back the LCP. Adding it as SVG in the markup could boost the score further
- The most major part is TBT. This suggest that there is major main thread work going on. Do you happen to know if there is any extensive upfront going on?

**Comment by BernardA** [comment]
@prateekbh, thank you for taking the time.

Things are NOT progressing well. 

I have made a few changes:
- upgraded to 9.5.3
- added dynamic imports for elements below the folder
- corrected image size
- change images to svg and self host it ( previously hosted at API domain )
- self host fonts

The only improvement was on the CLS ( Cumulative Layout Shift ), by correcting the image size that caused items to reshuffle. Even for that one I am not getting consistent readings. Sometimes it shows a zero, sometimes as a penalty.

As you can see below the main culprit is still the LCP ( Largest Contentful Paint ). This is quite odd considering that the image in question - now in SVG and self hosted - **does not weigh more than 2K**.

According to the [Lighthouse docs](https://web.dev/lcp/) , LCP is primarily affected by four factors:

>         Slow server response times
>         Render-blocking JavaScript and CSS
>         Resource load times
>         Client-side rendering

I cannot control the server and none of the other items seem to be an issue. So I am at a bit of a loss on why this turns out as bad as it does.

Any ideas?
![Screenshot 2020-09-03 at 21 31 19](https://user-images.githubusercontent.com/9449665/92160167-e7051e00-ee2e-11ea-9a50-3a13ad707ede.png)

**Reply by idiglove** [reply]
I'm getting `Remove unused Javascript` for mobile lighthouse too, and I dont know how to fix. Any ideas guys?

**Reply by abhikumar22** [reply]
Same here.

Nextjs empty app have low lighthouse score in mobile. What's the issue here ???

**Reply by ckruppe** [reply]
Maybe some good read-in for mobile lighthouse:
https://github.com/GoogleChrome/lighthouse/issues/7017

TLDR: don't take it to seriously.
